src/USER-LD/pair_localdensity.cpp

```diff
@@ -0,0 +1,883 @@
+        if (a[k][jtype]) index = 2;
+        
+        if (LD[k][index] <= rho_min[k]) {
+            coeff = frho_spline[k][0];
+            dFdrho = coeff[2];
+            uLD +=  a[k][itype] * ( coeff[6] + dFdrho*(LD[k][index] - rho_min[k]) );
+        }
+        else if (LD[k][index] >= rho_max[k]) {
+            coeff = frho_spline[k][nrho-1];
+            dFdrho = coeff[0] + coeff[1]  + coeff[2];
+            uLD +=  a[k][itype] * ( (coeff[3] + coeff[4] + coeff[5] + coeff[6]) + dFdrho*(LD[k][index] - rho_max[k]) );
+        }
+        else {
+            p = (LD[k][index] - rho_min[k]) / delta_rho[k];
+            m = static_cast<int> (p);
+            m = MAX(0, MIN(m, nrho-2));
+            p -= m;
+            p = MIN(p, 1.0);
+            coeff = frho_spline[k][m];
+            dFdrho = (coeff[0]*p + coeff[1])*p + coeff[2];
+            uLD +=  a[k][itype] * (((coeff[3]*p + coeff[4])*p + coeff[5])*p + coeff[6]);
+        }
+
+        if (rsq < lowercutsq[k]) {
+           dphi = 0.0;
+        }
+        else if (rsq > uppercutsq[k]) {
+           dphi = 0.0;
+        }
+        else {
+           dphi = rsq * (2.0*c2[k] + rsq * (4.0*c4[k] + 6.0*c6[k]*rsq));
+        }
+        fforce +=  -(a[k][itype]*b[k][jtype]*dFdrho + a[k][jtype]*b[k][itype]*dFdrho) * dphi *rsqinv;
+    }
+    memory->destroy(LD);
+
+    return uLD;
+}
+
+/*--------------------------------------------------------------------
+   spline the array frho read in from the file to create
+   frho_spline
+---------------------------------------------------------------------- */
+
+void PairLOCALDENSITY::array2spline() {
+  memory->destroy(frho_spline);
+  memory->create(frho_spline, nLD, nrho, 7, "pairLD:frho_spline");
+
+  for (int k = 0; k < nLD; k++)
+    interpolate_cbspl(nrho, delta_rho[k], frho[k], frho_spline[k]);
+
+}
+
+/* ---------------------------------------------------------------------- 
+  (one-dimensional) cubic spline interpolation sub-routine, 
+  which determines the coeffs for a clamped cubic spline 
+  given tabulated data
+ ------------------------------------------------------------------------*/
+
+void PairLOCALDENSITY::interpolate_cbspl(int n, double delta, 
+                                         double *f, double **spline) 
+{
+/*   inputs:
+          n     number of interpolating points
+	
+          f		array containing function values to
+			    be interpolated;  f[i] is the function
+			    value corresponding to x[i]
+				('x' refers to the independent var)
+	 
+	 delta     difference in tabulated values
+	           of x
+     
+	 outputs: (packaged as columns of the coeff matrix)
+          b		coeffs of linear terms
+	      c		coeffs of quadratic terms
+	      d		coeffs of cubic terms
+     spline     matrix that collects b,c,d
+     
+	 
+	 other parameters:
+          fpa		derivative of function at x=a
+          fpb		derivative of function at x=b
+*/
+                     
+     double *dl, *dd, *du;
+     double *b, *c, *d;
```

Comment:
the pointer variable `b` here shadows the class member pointer variable `b`. One of them should be renamed.